pkg/artifactory/datasource/datasource_artifactory_group.go

```diff
@@ -0,0 +1,62 @@
+package datasource
+
+import (
+	"fmt"
+	"github.com/go-resty/resty/v2"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/jfrog/terraform-provider-artifactory/v6/pkg/artifactory/resource/security"
+	"github.com/jfrog/terraform-provider-shared/packer"
+	"github.com/jfrog/terraform-provider-shared/predicate"
+	"github.com/jfrog/terraform-provider-shared/util"
+	"golang.org/x/net/context"
+	"net/http"
+)
+
+func ArtifactoryGroupSchema() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: DataSourceGroupRead,
+		Schema:      getDataSourceGroupSchema(),
+		Description: "Provides the Artifactory Group data source. Contains information about the group configuration and optionally provides its associated user list.",
+	}
+}
+
+func getDataSourceGroupSchema() map[string]*schema.Schema {
+	var includeUsersSchema = map[string]*schema.Schema{
+		"include_users": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Default:     false,
+			Description: "Setting includeUsers to true will return the group with its associated user list attached.",
+		},
+	}
+
+	baseGroupSchema := security.GroupSchema
+	delete(baseGroupSchema, "detach_all_users")
+	return util.MergeMaps(baseGroupSchema, includeUsersSchema)
+}
+
+func DataSourceGroupRead(_ context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	group := security.Group{}
+	name := d.Get("name").(string)
+	includeUsers := d.Get("include_users").(string)
+	url := fmt.Sprintf("%s%s?includeUsers=%s", security.GroupsEndpoint, name, includeUsers)
+	resp, err := m.(*resty.Client).R().SetResult(&group).Get(url)
+
+	if err != nil {
+		if resp != nil && (resp.StatusCode() == http.StatusBadRequest || resp.StatusCode() == http.StatusNotFound) {
```

Comment:
Not sure this is necessary at all.

Reply:
You don't think we need error handling if the user tries to get a group that doesn't exist?

Reply:
We already return diag.FromErr() if there's any error from the request. This extra check is not necessary for data source. For resource, this extra code is to reset TF ID in the state.